core/src/gossip_service.rs

```diff
@@ -115,6 +117,16 @@ pub fn get_clients(nodes: &[ContactInfo]) -> Vec<ThinClient> {
         .collect()
 }
 
+/// Creates a ThinClient by selecting a valid node at random
+pub fn get_client(nodes: &[ContactInfo]) -> ThinClient {
```

Comment:
Do you think this will be used outside of `core/src/replicator.rs`?  If not let's keep this functionally in there

Reply:
It's pretty useful, I don't know what else is using "discover->make a client" but this is a very fair way to pick a node to punish with txs or rpc. We've already got  bench tps and bench exchange using the "many" function.

Reply:
Maybe bench-{tps,exchange} can switch over to this, they have their own impl?

Naming nit on `get_client()` though.  Perhaps `get_random_client`?

Reply:
Those two uses need multiple clients (for multiple parallel transfers). I'm not attached to the name but I felt get_client was fair since the caller has no way to choose which node in particular it wants to connect to anyway. 
It's just saying "hey here's a bunch of nodes, gimme a client".  If that justification isn't good enough I'll switch the name :D